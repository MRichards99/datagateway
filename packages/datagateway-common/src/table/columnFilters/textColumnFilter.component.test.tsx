import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import TextColumnFilter from './textColumnFilter.component';

describe('Text filter component', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'div' });
    mount = createMount();

    jest.useRealTimers();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <TextColumnFilter label="test" onChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the onChange method once when input is typed and calls again by debounced function after timeout', (done) => {
    const onChange = jest.fn();

    const wrapper = mount(
      <TextColumnFilter label="test" onChange={onChange} />
    );

    // We simulate a change in the input from 'test' to 'test-again'.
    const textFilterInput = wrapper.find('input');

    textFilterInput.instance().value = 'test';
    textFilterInput.simulate('change');

    textFilterInput.instance().value = 'test-again';
    textFilterInput.simulate('change');

    // Jest timers do not co-operate with lodash.debounce, hence we use jest.useRealTimers
    // and this setTimeout function set to the actual delay of the debouncing function.
    setTimeout(() => {
      expect(onChange).toHaveBeenCalledTimes(1);
      expect(onChange).toHaveBeenLastCalledWith('test-again');
      done();
    }, 250);
  });
});
