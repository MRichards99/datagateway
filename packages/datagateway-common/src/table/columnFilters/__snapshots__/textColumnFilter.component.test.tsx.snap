// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text filter component renders correctly 1`] = `
<div>
  <WithStyles(ForwardRef(Input))
    aria-label="Filter by test"
    id="test filter"
    onChange={[Function]}
    placeholder="Contains..."
    value=""
  />
</div>
`;
